### PR TITLE
Nc news 4

### PR DESCRIPTION
**Pull request by JoeAshcroft**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -42,6 +42,44 @@ describe("GET /api", () => {
...
 });
 
+describe("GET /api/articles/:article_id", () => {
+  test("Should return a status of 200 and repsond with an object containing the relevant article based on its id", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article).toMatchObject({
+          article_id: expect.any(Number),
```
Comment (corkesw) [inline __tests__/app.test.js:52]: You should assert that the article_id is actually 1 here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -42,6 +42,44 @@ describe("GET /api", () => {
...
+          created_at: expect.any(String),
+        });
+      });
+  });
+  test("should respond with 400 error when article_id is an invalid type", () => {
+    return request(app)
+      .get("/api/articles/notanumber")
+      .expect(400)
+      .then(({ body }) => {
+        console.log(body);
```
Comment (corkesw) [inline __tests__/app.test.js]: get rid of the console.logs before merging

File: __tests__/app.test.js [hunk 3]
```diff
@@ -42,6 +42,44 @@ describe("GET /api", () => {
...
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("Should respond with 404 Not Found if given valid article_id but which has no corresponding article", () => {
+    return request(app)
+      .get("/api/articles/9999")
+      .expect(404)
+      .then(({ body }) => {
+        console.log(body);
```
Comment (corkesw) [inline __tests__/app.test.js]: one more here!

File: db/controller/app.controller.js [hunk 4]
```diff
@@ -20,4 +20,15 @@ const searchEndpoints = (req, res) => {
...
+  const { article_id } = req.params;
+  getArticleById(article_id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
```
Comment (corkesw) [inline db/controller/app.controller.js:38]: controller looks good :)

File: db/model/app.model.js [hunk 5]
```diff
@@ -1,11 +1,20 @@
...
+const getArticleById = (article_id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
+      return rows[0];
+    });
+};
```
Comment (corkesw) [inline db/model/app.model.js:19]: model looks good :)

File: db/app.js [hunk 6]
```diff
@@ -3,19 +3,33 @@ const app = express();
...
+app.get("/api/articles/:article_id", searchArticleById);
+
 app.all("*", (_, res) => {
   res.status(404).send({ status: 404, msg: "Not found" });
 });
 
 app.use((err, req, res, next) => {
-  return res.status(500).send({ mgs: "Internal Server Error" });
+  if (err.msg) {
+    res.status(404).send({ msg: err.msg });
```
Comment (corkesw) [inline db/app.js:24]: This works fine here so doesn't need to change but if you had a custom error that needed a dynamic status code, you are already passing the code on the `thrown` error object so you can access it there

**Comment by corkesw** [comment]
Hey Joe, looks good but please fix that assertion in the test. Be sure to remove any logs too. Once you've actioned these points, you are good to merge. No need to resubmit :)

**Comment by JoeAshcroft** [comment]
Will update assertion and console logs already gone - I noticed them the moment I submitted!
Thanks for the swift response



Sent from my Galaxy



-------- Original message --------
From: corkesw ***@***.***>
Date: 28/06/2023 15:46 (GMT+00:00)
To: KeeseRex/news-project ***@***.***>
Cc: Joe Ashcroft ***@***.***>, Author ***@***.***>
Subject: Re: [KeeseRex/news-project] Nc news 4 (PR #4)


Hey Joe, looks good but please fix that assertion in the test. Be sure to remove any logs too. Once you've actioned these points, you are good to merge. No need to resubmit :)

**Pull request merged**